lib/Dialect/LWE/IR/LWEOps.td

```diff
@@ -147,6 +147,36 @@ def LWE_RLWEDecodeOp : LWE_Op<"rlwe_decode", [
   let assemblyFormat = "$input attr-dict `:` qualified(type($input)) `->` qualified(type($output))";
 }
 
+def LWE_RLWEEncryptOp : LWE_Op<"rlwe_encrypt", [Pure,
+    EncodingsMatch<"input", "RLWEPlaintextType", "output", "RLWECiphertextType">]> {
+  let summary = "Encrypt an RLWE plaintext to a RLWE ciphertext";
+  let description = [{
+    Encrypt an RLWE plaintext to yield a RLWE ciphertext.
+  }];
+
+  let arguments = (ins
+    RLWEPlaintext:$input,
+    RLWESecretOrPublicKey:$key
+  );
+  let results = (outs RLWECiphertext:$output);
+  let hasVerifier = 1;
+}
+
+def LWE_RLWEDecryptOp : LWE_Op<"rlwe_decrypt", [Pure,
+    EncodingsMatch<"input", "RLWECiphertextType", "output", "RLWEPlaintextType">]> {
```

Comment:
like mentioned above, can you add `RlweParametersMatch<"secret_key", "RLWESecretKeyType", "input", "RLWECiphertextType">` please?

Reply:
Changed and done.